### DISCUSSION TITLE
Generating files for separated grammar

### DISCUSSION THREAD

**Discussion by blin04** [antlr/antlr4#3225] [Q&A]
Hello!
I have a probably very simple question, but for some reason I just can't find a solution to it. I am trying to parse a C++ code for my project (I'm building a transpiler). The grammar for C++ is separated, namely it doesn't have a single .g4 file but it has two of them, CPP14Lexer.g4 and CPP14Parser.g4. How can I generate parser this way? When I used a singe .g4 file, I did it with this command in my command prompt:
antlr4 -Dlanguage=Python3 -visitor GrammarName.g4
Thank you very much for your help!

**Comment by mildmojo** [comment]
Aha! There's two of us! I also struggled to find this documented anywhere, even in off-site tutorials! After a lot of searching, I started experimenting, and your command should look something like:

```
antlr4 -Dlanguage=Python3 -visitor GrammarName.g4 LexerName.g4
```

To build a split grammar with ANTLR, you pass both the lexer and parser grammar files as arguments to `antlr4`.

When providing split grammar files, ANTLR seems to pass the filenames through to the parser/lexer output files, so it won't tack on `Parser` anymore. I renamed my grammar files `langParser.g4` and `langLexer.g4`.

**Reply by kaby76** [reply]
Notes:
* Install [this toolkit](https://github.com/antlr/antlr4-tools). It will make your life easier.
* To generate the parser, type `antlr4 -Dlanguage=Python3 *.g4`.
* [grammars-v4/cpp](https://github.com/antlr/grammars-v4/tree/master/cpp) is not "target agnostic". [It is not tested for Python3](https://github.com/antlr/grammars-v4/blob/128ebeb55f66061675726ce35e0f47c5c6bfb9d0/_scripts/skip-python3.txt#L9). You'll have to clean up the [pureSpecifier rule](https://github.com/antlr/grammars-v4/blob/128ebeb55f66061675726ce35e0f47c5c6bfb9d0/cpp/CPP14Parser.g4#L646).
